container/concurrent/map.go

```diff
@@ -34,13 +34,13 @@ func (p *mapImp) Get(key interface{}) (value interface{}, found bool) {
 
 func (p *mapImp) Set(key interface{}, value interface{}) {
 	p.lock.Lock()
-	p.lock.Unlock()
+	defer p.lock.Unlock()
```

Comment:
Am not familiar with this code but how is this even working?! Would be good to add a test that exercises concurrent updates to avoid future regression.